pkg/kubelet/logs/container_log_manager.go

```diff
@@ -171,20 +173,28 @@ func NewContainerLogManager(runtimeService internalapi.RuntimeService, osInterfa
 			MaxSize:  parsedMaxSize,
 			MaxFiles: maxFiles,
 		},
-		clock: clock.RealClock{},
-		mutex: sync.Mutex{},
+		clock:            clock.RealClock{},
+		mutex:            sync.Mutex{},
+		maxWorkers:       maxWorkers,
+		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "kubelet_log_rotate_manager"),
+		monitoringPeriod: monitorDuration,
 	}, nil
 }
 
 // Start the container log manager.
 func (c *containerLogManager) Start() {
 	ctx := context.Background()
+	klog.InfoS("Initializing container log rotate workers", "workers", c.maxWorkers, "monitorPeriod", c.monitoringPeriod)
+	for i := 0; i <= c.maxWorkers; i++ {
+		worker := i + 1
+		go c.processQueueItems(worker)
```

Comment:
Similar to how we pass in `ctx` to rotateLogs below, can we please add `ctx` as a parameter into `processQueueItems` as well? (then have the infinite loop terminate based on the context?

Reply:
( will be useful in the future when we figure out when/how to stop both this processing queue and rotate logs - say when the kubelet is terminating? )

Reply:
`processQueueItems` can pass it down into `processContainer` instead of using `context.Background()`

Reply:
Ack. That can be done. Let me take care of that and push the changes.

Reply:
Done

Reply:
Pushed the changes a little too early and broke the tests.. Fixing it now